### PR TITLE
IGNITE-16021 Update Apache Ignite 2.12 release notes

### PR DESCRIPTION
**Pull request by NSAmelchev**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: RELEASE_NOTES.txt [hunk 1]
```diff
@@ -1,6 +1,141 @@
 Apache Ignite Release Notes
 ===========================
 
+Apache Ignite In-Memory Distributed Database 2.12.0
+-----------------------------------------------------------
+
+(!) WARNINGS:
+* The community accepted deprecate for removal in future releases: CacheMode#LOCAL, CacheAtomicityMode#TRANSACTIONAL_SNAPSHOT, CacheConfiguration#rebalanceDelay.
```
Comment (Mmuzaf) [inline RELEASE_NOTES.txt:8]: We should mention that legacy service grid implementation is already removed in the next relese.

File: RELEASE_NOTES.txt [hunk 2]
```diff
@@ -1,6 +1,141 @@
...
+* Added support for creating IndexQuery without index name.
+* Added the ability to cancellation of consistency recovery command (Read Repair via control.ch).
+* Added the ability to record Control.sh consistency check violations to other log file.
+* Added the ability to restore snapshot taken on different topologies.
+* Added the ability to snapshot encrypted caches.
+* Added the cache destroy command for control.sh.
+* Added the force deactivation flag to the IgniteMXBean.
+* Added time metrics and statistics for the IgniteCache#getAllOutTx.
+* Added unconditional logging of tx states to WAL  to ensure correct tx recovery after node crash.
+* Any readRepair attempt now generates the event (if consistency violation found), even for atomic caches.
```
Comment (Mmuzaf) [inline RELEASE_NOTES.txt:37]:
```suggestion
* Added firing event for any ReadRepair attempt (if consistency violation found). Atomic caches are also supported.
```

File: RELEASE_NOTES.txt [hunk 3]
```diff
@@ -1,6 +1,141 @@
...
+.Net:
+* Fixed NRE in ClientFailoverSocket due to late logger setter.
+* Fixed NullPointerException in ContinuousQuery with security enabled.
+* Fixed Schema project version and examples packaging.
+* Fixed SslStreamFactory.CertificatePath null value.
+* Fixed TypeNameParser to ignore escaped characters in compiler-generated type names.
+* Fixed dynamic assemblies handling in TypeResolver.
+* Fixed thin client streamer not creating SQL table entries.
+* Fixed verify-nuget.ps1 failure when .NET 5 is installed.
+* Modernized project structure.
```
Comment (Mmuzaf) [inline RELEASE_NOTES.txt:121]: I don't think we should mention it in the release notes.

Comment (NSAmelchev) [reply RELEASE_NOTES.txt]: Fixed.

File: RELEASE_NOTES.txt (hunk 2 above)
Comment (NSAmelchev) [reply RELEASE_NOTES.txt]: Fixed.

File: RELEASE_NOTES.txt (hunk 1 above)
Comment (NSAmelchev) [reply RELEASE_NOTES.txt:8]: Fixed.

**Pull request closed without merging**